packages/lesswrong/components/posts/PostsEditForm.tsx

```diff
@@ -117,8 +126,15 @@ const PostsEditForm = ({ documentId, classes }: {
             if (options?.submitOptions?.redirectToEditor) {
               history.push(postGetEditUrl(post._id, false, post.linkSharingKey));
             } else {
-              history.push({pathname: postGetPageUrl(post)})
-              flash({ messageString: `Post "${post.title}" edited.`, type: 'success'});
+              // If they are publishing a draft, show the share popup
+              // Note: we can't use isDraft here because it gets updated to true when they click "Publish"
+              const showSharePopup = isEAForum && wasEverDraft.current && !post.draft
+              const sharePostQuery = `?${SHARE_POPUP_QUERY_PARAM}=true`
+              history.push({pathname: postGetPageUrl(post), search: showSharePopup ? sharePostQuery : ''})
```

Comment:
I experimented with different ways of getting the popup to open, including trying to do an `openDialog` here. I ended up going with adding a query parameter, which then gets removed after the first render in `PostsPage`. Some benefits of doing it this way:
 - No fiddling around with the `closeOnNavigate` setting to make it open in this page, stay open on the post page, but then close if you navigate away from the post page
 - The animation is synced to the loading of the post page
 - You can test it quickly by adding the query param without going through the full edit cycle
 - Removing the query parameter after loading means you won't accidentally share a link that has the popup open